### ISSUE TITLE
Bug: Global user role null verb causes interface to become uneditable

### ISSUE THREAD

**Issue by patrick0057** [rancher/rancher#28040]
**What kind of request is this (question/bug/enhancement/feature request):**
Bug

**Description:**
Having verbs set to null in the global user role causes the whole role to become uneditable. 
```
{
"apiGroups": [
"*"
],
"resources": [
"persistentvolumes"
],
"type": "/v3/schemas/policyRule",
"verbs": null
}
]
```
![image](https://user-images.githubusercontent.com/44879867/88302751-10bd1800-ccbb-11ea-8100-d785afab6bdb.png)
gz#11380

**Comment by sowmyav27** [comment]
**Reproduced the issue on `2.4.5`**

- Clone a global role `Create Clusters`  to say `test-1`
- View in API of the role--> Edit role --> Change a rule's verb to null

```
"apiGroups": [
"management.cattle.io"
],
"resources": [ 2 items
"templates",
"templateversions"
],
"type": "/v3/schemas/policyRule",
"verbs": null
},
```

- Click on Send request. The changes are saved.
- From Rancher UI --> Edit Role `test-1`
- The UI is not editable.
- Error in Console:

```
component.js:36 Uncaught TypeError: Cannot read property 'indexOf' of null
    at component.js:36
    at Array.map (<anonymous>)
    at n.init (component.js:33)
    at n [as init] (utils.js:365)
    at b (core_object.js:89)
    at Function.e.create (core_object.js:686)
    at e.n.create (container.js:554)
    at t.a.create (glimmer.js:5538)
    at Object.evaluate (runtime.js:1569)
    at e.t.evaluate (runtime.js:73)
runtime.js:3847 Uncaught TypeError: Cannot read property 'lastNode' of null
    at e.t.lastNode (runtime.js:3847)
    at n.t.lastNode (runtime.js:3847)
    at i.r.lastNode (runtime.js:4367)
    at e.t.lastNode (runtime.js:3973)
    at i.a.evaluate (runtime.js:4551)
    at e.r.execute (runtime.js:4308)
    at e.t.rerender (runtime.js:4635)
    at e.s.render (glimmer.js:5994)
    at V (metal.js:1006)
```
<img width="1915" alt="Screen Shot 2020-07-23 at 9 18 05 AM" src="https://user-images.githubusercontent.com/26032343/88311187-6f878f00-ccc5-11ea-96ab-e3f461def232.png">

**Comment by sadiapoddar** [comment]
Couldn't repro with the version 2.4.5
Steps used to repro
1. Created a clone of a cluster role
2. Edited the cluster role in "View API" and set the verb to null
<img width="297" alt="Screen Shot 2020-11-19 at 5 55 49 PM" src="https://user-images.githubusercontent.com/72409237/99748855-2e946100-2a92-11eb-84e8-60768bb56e1d.png">

3. Send the request, the request didn't set and it shows the value can't be null and this is an invalid request. 
<img width="494" alt="Screen Shot 2020-11-19 at 5 56 20 PM" src="https://user-images.githubusercontent.com/72409237/99748887-3fdd6d80-2a92-11eb-82ab-8b67d949014c.png">

**Comment by sadiapoddar** [comment]
I tried to repro this doing the following steps
1. select global
2. create role
3. clone "create cluster" role
4. edit using API view
5. send the request and seeing https 422, meaning error in request format.

**Comment by sadiapoddar** [comment]
Verified the fix on 2.5-head, 9ee39b9, and master-head, a549b47.
Verification steps.

1. select global
2. create a role
3. clone "create cluster" role
4. edit using API view
5. send the request and seeing HTTP 200, status was ok.

<img width="217" alt="Screen Shot 2020-11-30 at 3 04 46 PM" src="https://user-images.githubusercontent.com/72409237/100676529-f0aefc80-331d-11eb-93d2-a91b05933cd4.png">
<img width="205" alt="Screen Shot 2020-11-30 at 3 04 23 PM" src="https://user-images.githubusercontent.com/72409237/100676547-f6a4dd80-331d-11eb-8ad9-a46a6634148a.png">
6. Re-Edit doesn't hang and there is no exception on the console

<img width="1502" alt="Screen Shot 2020-11-30 at 3 10 11 PM" src="https://user-images.githubusercontent.com/72409237/100676857-8f3b5d80-331e-11eb-9bd6-2d4787cdd020.png">

**Comment by sadiapoddar** [comment]
Verified the fix on 2.5-head, 9ee39b9, and master-head, a549b47.
Verification steps.

1. select global
2. create a role
3. clone "create cluster" role
4. edit using API view
5. send the request and seeing HTTP 200, status was ok.

<img width="217" alt="Screen Shot 2020-11-30 at 3 04 46 PM" src="https://user-images.githubusercontent.com/72409237/100676529-f0aefc80-331d-11eb-93d2-a91b05933cd4.png">
<img width="205" alt="Screen Shot 2020-11-30 at 3 04 23 PM" src="https://user-images.githubusercontent.com/72409237/100676547-f6a4dd80-331d-11eb-8ad9-a46a6634148a.png">
6. Re-Edit doesn't hang and there is no exception on the console

<img width="1502" alt="Screen Shot 2020-11-30 at 3 10 11 PM" src="https://user-images.githubusercontent.com/72409237/100676857-8f3b5d80-331e-11eb-9bd6-2d4787cdd020.png">

**Comment by sadiapoddar** [comment]
Closing the issue and as it's fixed.
